### ISSUE TITLE
Picoseconds are inconsistently formatted

### ISSUE THREAD

**Issue by proegssilb** [proegssilb/ferris-elf#69]
One response I got from the bot:

> Benchmark complete (Unverified)
> Median: 2282817689.5
> Average: 2333516918.5

Another:

> Benchmark complete (Unverified)
> Median: 498.39µs
> Average: 507133367.5

Let's write up some test cases and get this behaving predictably.

**Comment by ultrabear** [comment]
This is probably caused by a rouge cast/type ignore somewhere (or the most innocuous one: arithmetic), as those fields should be typed as Picoseconds and Picoseconds has strict formatting, and its repr version also says Picoseconds

**Comment by ultrabear** [comment]
ohhhhh
statistics.mean
thanks subclasses

**Comment by proegssilb** [comment]
We can solve this later, especially since IMO, this issue needs to be a catalyst for getting some unit tests in order. In the vein of "creative testing", it'd be amazing if the second option could be verified with testing somehow.

**Comment by proegssilb** [comment]
It's actually worse than just making the types spit out a Picoseconds object. It internally works in Fraction, and then does this as a final conversion step:

```python
def _convert(value, T):
    """Convert value to given numeric type T."""
    if type(value) is T:
        # This covers the cases where T is Fraction, or where value is
        # a NAN or INF (Decimal or float).
        return value
    if issubclass(T, int) and value.denominator != 1:
        T = float
    try:
        # FIXME: what do we do if this overflows?
        return T(value)
    except TypeError:
        if issubclass(T, Decimal):
            return T(value.numerator) / T(value.denominator)
        else:
            raise
```

(at least, on my machine, running 3.12)

Because we subclass `int`, I expect this code to return `float` more often than not. Not sure why the inconsistent results, but the results are inconsistent.

**Comment by proegssilb** [comment]
Fixed by #74 , just forgot to link them.
